### PR TITLE
added a movie details page with movie cast

### LINKED ISSUE #13: Add a 'Movie Details' Page
### Expected behavior

Add a page that allows users to read the movie description and title on their own, with a list of movies displayed on the movie_list page. When a user clicks on a movie title, they should be taken to a separate movie_details page where all the details of the selected movie, such as the title, description, release date, runtime (if available), are displayed. Additionally, the cast information should be fetched from the OMDb API, which is already used in the add_movie view, and displayed on the movie_details page in a clear format. The cast should be presented as a list, with each actor's name displayed on a new line.
DON'T add movie ratings.
image

### Current behavior
Right now clicking on the movie in the movie_list doesn't take us to a new page.
image

### PR DESCRIPTION
**Pull request by hexwhiz**
## Description

Added some fields in the movie model like **(runtime, release date etc.)**. These fields will be populated while adding movie through add_movie view. Added a movie details page that show the details of the movie in which user clicks. While rendering the movie details page an API call to TMDB database is made to fetch the cast list of the movie. An API key is needed for the TMDB API that is saved in env file. [Here](https://developer.themoviedb.org/docs/getting-started) is the documentation to get the key from TMDB. 

## Related Issue

Fixes #13 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Motivation and Context

This change was required because a movie booking website should have the details of movies.

## How Has This Been Tested?

I have tested the new functionality by adding some movies and then clicking on the poster in movie list. Here is a video showing the same. 

## Screenshots (if appropriate):
https://github.com/user-attachments/assets/ca71ecc3-2b9c-447d-8886-6f7c45d3bf6e

## Checklist:

- [x] I have registered myself at [Contrihub website](https://sac.mnnit.ac.in/contrihub).
- [x] My code follows the code style of this project.
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

### CODE REVIEW

**Comment by eleensmathew** [comment]
Can you fix the conflicts

**Comment by hexwhiz** [comment]
@eleensmathew ya let me see

**Comment by hexwhiz** [comment]
@eleensmathew conflicts have been resolved. You can review the PR.

**Comment by eleensmathew** [comment]
I think in most movie related sites they add the images of the cast as well. Can you try implementing that also

**Comment by hexwhiz** [comment]
Ok I will add that. I was thinking to add that but PR timer was running out so I skipped that.

**Comment by hexwhiz** [comment]
@eleensmathew done adding cast images.
---

image

File: user/views.py [hunk 1]
```diff
@@ -22,12 +22,51 @@
...
+            croplen=0
+            if len(data['cast'])>=10:
+                croplen=10
+            else:
+                croplen=len(data['cast'])
+            filtered_cast = [
+                {
+                    'name': member['name'],
+                    'character': member['character'],
+                    'img_url' : f'https://image.tmdb.org/t/p/w200{member['profile_path']}'
```
Comment (eleensmathew) [inline user/views.py]: I am getting the following error 
`'img_url' : f'https://image.tmdb.org/t/p/w200{member['profile_path']}'
                                                          ^^^^^^^^^^^^
SyntaxError: f-string: unmatched '['`
could you change it to?
` 'img_url' : f"https://image.tmdb.org/t/p/w200{member['profile_path']}"`

**Comment by eleensmathew** [comment]
Add information in readme how to get tmdb api key and add it in .env file

**Comment by eleensmathew** [comment]
Everything else looks good so please make the requested changes.

**Comment by hexwhiz** [comment]
Ok I will make changes.

**Comment by hexwhiz** [comment]
@eleensmathew said changes have been made

**Pull request merged**